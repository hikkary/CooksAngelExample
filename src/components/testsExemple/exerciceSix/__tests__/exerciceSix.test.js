import React from 'react';
import ExerciceSix from '../';

import { shallow } from 'enzyme';

var localStorageMock = (() => {
  var store = {};
  return {
    getItem: key => {
      return store[key];
    },
    setItem: (key, value) => {
      store[key] = value.toString();
    },
    clear: () => {
      store = {};
    },
    removeItem: key => {
      delete store[key];
    }
  };
})();

Object.defineProperty(window, 'localStorage', { value: localStorageMock });

// console.log(localStorage)
beforeEach(() => {
  localStorage.clear();
});

it('should match the snapshot with props', () => {
  const component = shallow(<ExerciceSix data="test" />);

  expect(component).toMatchSnapshot();
});

it('should match the snapshot without props', () => {
  const component = shallow(<ExerciceSix />);

  expect(component).toMatchSnapshot();
});

it('should set the localStorage to the given string', () => {
  shallow(<ExerciceSix data="coco" />);

  expect(localStorage.getItem('data')).toBe('coco');
});

it('should set the localStorage to the given long string', () => {
  // console.log(localStorage.getItem('data'))
  // on se rend compte ici que la data est toujours dans le local storage si on utilise pas before Each
  shallow(<ExerciceSix data="cocococococococococococococo" />);

  expect(localStorage.getItem('data')).toBe('cocococococococococococococo');
});

it('should remove item called data set in the localStorage ', () => {
  shallow(<ExerciceSix data="test" />);
  localStorage.removeItem('data');

  expect(localStorage.getItem('data')).toBe(undefined);
});

it('should clear all data set in the localStorage ', () => {
  shallow(<ExerciceSix data="test" />);
  localStorage.clear();

  expect(localStorage.getItem('data')).toBe(undefined);
});
