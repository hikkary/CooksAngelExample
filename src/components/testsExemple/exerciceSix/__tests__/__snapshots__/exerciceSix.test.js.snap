// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot with props 1`] = `
<div
  className="LocalStorageContainer"
>
  <p>
    test
  </p>
</div>
`;

exports[`should match the snapshot without props 1`] = `
<div
  className="LocalStorageContainer"
>
  <p>
    test
  </p>
</div>
`;
